tests/kohana/ArrTest.php

```diff
@@ -416,6 +416,8 @@ public function provider_path()
 			'object' => new ArrayObject(array('iterator' => TRUE)), // Iterable object should work exactly the same
 		);
 
+		$array_object = new ArrayObject($array);
```

Comment:
Prob want to do this recursively / create it manually so that the nested levels are also ArrayObject just to be sure we eliminate any current/future oddity below the top-level?

Reply:
See 599ffdd. Created manually so its hopefully clearer to see why than doing it recursively